cluster/gce/util.sh

```diff
@@ -732,35 +732,33 @@ function setup-monitoring {
     echo "Setting up cluster monitoring using Heapster."
 
     detect-project
-    if ! gcloud compute firewall-rules --project "{$PROJECT}" describe monitoring-heapster &>/dev/null; then
+    if ! gcloud compute firewall-rules --project "${PROJECT}" describe monitoring-heapster &> /dev/null; then
       if ! gcloud compute firewall-rules create monitoring-heapster \
           --project "${PROJECT}" \
           --target-tags="${MINION_TAG}" \
           --network="${NETWORK}" \
-          --allow tcp:80 tcp:8083 tcp:8086 tcp:9200; then
-        echo "Failed to set up firewall for monitoring" && false
+          --allow tcp:80 tcp:8083 tcp:8086; then
+        echo -e "${color_red}Failed to set up firewall for monitoring ${color_norm}" && false
       fi
     fi
 
-    # Re-use master auth for Grafana
-    get-password
-    ensure-temp-dir
-
-    sed -e "s/{KUBE_USER}/$KUBE_USER/g" \
-        -e "s/{KUBE_PASSWORD}/$KUBE_PASSWORD/g" \
-        "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influx-grafana-pod.json" \
-      > "${KUBE_TEMP}/influx-grafana-pod.json"
     local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
-    if "${kubectl}" create -f "${KUBE_TEMP}/influx-grafana-pod.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influx-grafana-service.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/heapster-pod.json" &> /dev/null; then
-      local dashboard_url="http://$(${kubectl} get -o json pod influx-grafana | grep hostIP | awk '{print $2}' | sed 's/[,|\"]//g')"
+    if "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influxdb-grafana-controller.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influxdb-service.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/heapster-controller.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/grafana-service.yaml" &> /dev/null; then
```

Comment:
I believe you can now pass the entire directory to kubectl and it will make all at the same time?